spec/unit/file_system_spec.rb

```diff
@@ -5,6 +5,18 @@
 describe "Puppet::FileSystem" do
   include PuppetSpec::Files
 
+  def with_file_content(content)
+    path = tmpfile('file-system')
+    file = File.new(path, 'wb')
+    file.sync = true
+    file.print content
+
+    yield path
+
+  ensure
+    file.close
```

Comment:
One thing to watch out for is that `file` could be nil, e.g. if `tmpfile` or `File.new` raise an error. I generally nest the begin/rescue at the point where you know you have a valid file to close, e.g.

```
path = tmpfile('file-system')
file = File.new(path, 'wb')
begin
  ...
ensure
  file.close
end
```